### PR TITLE
Updaing CQConfigManagerClient to add wait for template, policies, templateTypes Path to exist

### PR DESCRIPTION
**Pull request by absarasw**

### CODE REVIEW

**Comment by absarasw** [comment]
@dulvac , @volteanu Can you please review and merge the PR?

**Comment by absarasw** [comment]
@volteanu , @dulvac, Can you please review and merge this PR?

File: src/main/java/com/adobe/cq/testing/client/security/ExtendedCQPermissions.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+     * @return Json node containing resulting permissions
+     * @throws ClientException
+     *          If something fails during request/response cycle after 10 times unsuccessful retry
+     */
+    public JsonNode changePermissionsWithRetry(PermissionConfig config, int... expectedStatus) throws ClientException {
+        int retries = 0;
+        int maxRetries = 10;
+        boolean passed = false;
+        JsonNode permissions = JsonNodeFactory.instance.objectNode();
+        while (!passed && (retries < maxRetries)) {
```
Comment (dulvac) [inline src/main/java/com/adobe/cq/testing/client/security/ExtendedCQPermissions.java]: Why not use the `Polling` utility from the sling testing clients?

File: src/main/java/com/adobe/cq/testing/client/security/ExtendedCQPermissions.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+            try {
+                permissions = super.changePermissions(config.getAuthorizableId(), config.getPath(),
+                        config.isRead(), config.isModify(), config.isCreate(), config.isDelete(),
+                        config.isAclRead(), config.isAclEdit(), config.isReplicate(), expectedStatus);
+                LOG.info("new set permission response string for {} at {}: {}",
+                        config.getAuthorizableId(), config.getPath(), permissions);
+                passed = true;
+            } catch (Exception e) {
+                if (retries + 1 == maxRetries) {
+                    LOG.error("Maximal retries reached!");
```
Comment (dulvac) [inline src/main/java/com/adobe/cq/testing/client/security/ExtendedCQPermissions.java]:
```suggestion
                    LOG.error("Maximum number of retries reached!");
```

**Review by dulvac — COMMENTED** [review]
@absarasw apologies for the very long delay and thanks for the PR!
I have a couple of minor comments, otherwise lgtm (I assume it's been tested)

File: src/main/java/com/adobe/cq/testing/client/security/ExtendedCQPermissions.java [hunk 3]
```diff
@@ -42,30 +45,37 @@
...
-            } finally {
-                retries++;
             }
         }
-        return permissions;
+
+        ChangePermissionsPolling createPolling = new ChangePermissionsPolling();
+        try {
+            createPolling.poll(timeout, delay);
+        } catch(TimeoutException e) {
```
Comment (dulvac) [inline src/main/java/com/adobe/cq/testing/client/security/ExtendedCQPermissions.java]:
```suggestion
        } catch (TimeoutException e) {
```

File: src/main/java/com/adobe/cq/testing/client/security/ExtendedCQPermissions.java [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+     * @param authorizableId the Id of the authorizable
+     * @param path           path
+     * @param depth          depth
+     * @param expectedStatus list of allowed HTTP Status to be returned. If not set,
+     *                       http status 200 (OK) is assumed.
+     * @return the root {@link JsonNode}
+     * @throws ClientException If something fails during request/response cycle after 10 times unsuccessful retry
+     */
+    @Override
+    public JsonNode getPermissions(String authorizableId, String path, int depth, int... expectedStatus) throws
```
Comment (dulvac) [inline src/main/java/com/adobe/cq/testing/client/security/ExtendedCQPermissions.java]: Same polling mechanism here as for create, please. Sorry that i missed this in the previous review.

**Comment by absarasw** [comment]
@dulvac , @volteanu , I have updated the PR with the review changes. Please review.

**Comment by dulvac** [comment]
Some minor javadoc problems, but I'll fix them separately. Thanks @absarasw

**Pull request merged**